src/jupyterlabcontroller/services/state.py

```diff
@@ -511,6 +511,49 @@ async def _clear_events(self, lab: UserLab) -> None:
         for trigger in triggers:
             trigger.set()
 
+    async def _reflect_events(
```

Comment:
I mean, I really do hate it that we've clawed our way back to the KubeSpawner reflector model, but I guess maybe they really did know what they were doing the first time.